code/modules/mob/living/simple_animal/slaughter/slaughter.dm

```diff
@@ -0,0 +1,274 @@
+	..(1)
+	new /obj/effect/decal/cleanable/blood (src.loc)
+	new /obj/effect/gibspawner/generic(get_turf(src))
+	new /obj/effect/gibspawner/generic(get_turf(src))
+	playsound(get_turf(src),'sound/misc/demon_dies.ogg', 200, 1)
+	new /obj/item/weapon/demonheart (src.loc)
+	visible_message("<span class='danger'>The [src] screams in anger as its form collapes into a pool of viscera.</span>")
+	ghostize()
+	qdel(src)
+	return
+
+
+
+////////////////////The Powers
+
+/mob/living/simple_animal/slaughter/proc/phaseout(var/obj/effect/decal/cleanable/B)
+	var/turf/mobloc = get_turf(src.loc)
+	var/turf/bloodloc = get_turf(B.loc)
+	if(Adjacent(bloodloc))
+		src.notransform = TRUE
+		spawn(0)
+			src.visible_message("The [src] sinks into [B].")
+			playsound(get_turf(src), 'sound/misc/enter_blood.ogg', 100, 1, -1)
+			var/obj/effect/dummy/slaughter/holder = new /obj/effect/dummy/slaughter( mobloc )
+			var/atom/movable/overlay/animation = new /atom/movable/overlay( mobloc )
+			animation.name = "odd blood"
+			animation.density = 0
+			animation.anchored = 1
+			animation.icon = 'icons/mob/mob.dmi'
+			animation.icon_state = "jaunt"
+			animation.layer = 5
+			animation.master = holder
+			animation.dir = src.dir
+			src.ExtinguishMob()
+			if(src.buckled)
+				src.buckled = null
+			if(src.pulling)
+				if(istype(src.pulling, /mob/living/))
+					var/mob/living/victim = src.pulling
+					var/mob/living/carbon/human/victimType = src.pulling
+					if (victimType.species.flags & IS_SYNTHETIC)//IPCs are too crunchy....I am also calling this wrong.
+						src.visible_message("The [src] lets go of [victim] at the last second!")
+						src.pulling = FALSE
+					else if(victim.stat == CONSCIOUS)
+						src.visible_message("[victim] kicks free of the [src] at the last second!")
+					else
+						victim.loc = holder///holder
+						src.visible_message("<span class='warning'><B>The [src] drags [victim] into [B]!</B>")
+						src.kidnapped = victim
+			flick("jaunt",animation)
+			src.loc = holder
+			src.phased = TRUE
+			src.holder = holder
+
+			if(src.kidnapped)
+				src << "<B>You being to feast on [kidnapped]. You can not move while you are doing this.</B>"
+				src.visible_message("<span class='warning'>Loud eating sounds come from the blood...</B>")
+				src.eating = TRUE
+				sleep(6)///wait till the anim finishs before we delet it..
+				del(animation)
+				new /obj/effect/gibspawner/generic(get_turf(src))
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				new /obj/effect/gibspawner/generic(get_turf(src))
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				new /obj/effect/gibspawner/generic(get_turf(src))
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				var/mob/living/carbon/human/nomNom = src.kidnapped
+				var/obj/item/organ/brain/brainObj = nomNom.internal_organs_by_name["brain"]
+				if(brainObj && istype(brainObj))
```

Comment:
@Fox-McCloud Well i wasn't sure if perma death would be alright or not. I can change it to try to throw out a random organ.